networking/routetable.go

```diff
@@ -176,44 +176,55 @@ func (table *routingTable) findPreferredRoute() (ok bool, ifname string, route *
 	return
 }
 
-func (table *routingTable) setPreferredRoute(removeDefaultRoute bool) (err error) {
+func (table *routingTable) setPreferredRoute(overrideDefaultRoute bool) (err error) {
 	table.lock.Lock()
 	// was their an old route, remove it.
-	if table.oldDefaultRoute != nil { //} && (table.oldDefaultRoute != table.defaultRoute) {
+	if table.oldDefaultRoute != nil && overrideDefaultRoute { //} && (table.oldDefaultRoute != table.defaultRoute) {
 		table.lock.Unlock()
 		netlink.RouteDel(table.oldDefaultRoute)
 		table.lock.Lock()
 	}
 	// update the route if it changed
 	//	if table.oldDefaultRoute != table.defaultRoute {
-	route := table.defaultRoute
-	table.lock.Unlock()
-	if route != nil {
-		err = netlink.RouteAdd(route)
-		if err != nil && err.Error() == "file exists" && removeDefaultRoute {
-			// get the link index for this interface
-			// _, ifindex, err2 := GetInterfaceIndexAndName(table.primaryDefaultRouteIf, 0)
-
-			// if err2 != nil {
-			// 	return
-			// }
-
-			existing := &netlink.Route{
-				Dst: &net.IPNet{
-					IP:   net.IPv4(0, 0, 0, 0),
-					Mask: net.IPv4Mask(0, 0, 0, 0),
-				},
-				// LinkIndex: ifindex,
-			}
-
-			err2 := netlink.RouteDel(existing)
-			if err2 != nil {
-				err = fmt.Errorf("Could not delete default route: %s", err2.Error())
-				return
-			}
-
+	if(overrideDefaultRoute) {
```

Comment:
What happens if a default route wasn't already configured and overrideDefaultRoute is false?  In other words, what is this function supposed to do if there is no default route configured on the system yet?

Reply:
Do you mean dont_overide_default_route??

Reply:
if overrideDefaultRoute is false, maestro will not set the default route. It is assumed in such scenarios, Network manager or some other service is handling the primary interface and will set the default route.

Reply:
Is that a correct assumption?  Wouldn't we want to set a default route if none is currently configured?

Reply:
We can change the code to do that. If no defaultroute is configured, maestro can add the default route of the interface it is handling.

Reply:
Assuming that no default route exists already, it might be good to set it.  Not sure what the criteria are for network-manager to override it later.

Reply:
If there are two default routes through two different interfaces in the routing table, the route with the lowest "Metric" is chosen. But the problem is maestro always sets the Metric to zero. Network Manager on the other hand I think is setting the metric to 100. So there is a possibility later on, network manager might not be able to override the default route.

Reply:
This is the reason I had to make sure maestro does not add a default route even if it is not deleting the default route that was set by network manager.